kotlin/workflow-host/src/test/java/com/squareup/workflow/internal/LifetimeTrackerTest.kt

```diff
@@ -90,4 +91,27 @@ class LifetimeTrackerTest {
     assertEquals(1, disposableCount)
     assertFalse("foo" in tracker.lifetimes)
   }
+
+  @Test fun `track throws on two duplicate keys`() {
+    val error = assertFailsWith<IllegalArgumentException> {
+      tracker.track(listOf("dup", "dup"))
+    }
+    assertTrue("Expected all keys to be unique. Duplicates:" in error.message!!)
+    assertTrue("2×dup" in error.message!!)
+  }
+
+  @Test fun `track throws on three duplicate keys`() {
+    val error = assertFailsWith<IllegalArgumentException> {
+      tracker.track(listOf("dup", "dup", "dup"))
+    }
+    assertTrue("3×dup" in error.message!!)
+  }
+
+  @Test fun `track throws on multiple sets of duplicate keys`() {
+    val error = assertFailsWith<IllegalArgumentException> {
+      tracker.track(listOf("dup1", "dup2", "dup1", "dup2"))
+    }
+    assertTrue("2×dup1" in error.message!!)
+    assertTrue("2×dup2" in error.message!!)
```

Comment:
nit: `assertNotNull` might help you avoid bangs (since it has nullability inside it) and give you a better error message in case `error.message` is `null`.
```kotlin
assertNotNull(error.message)
assertTrue("2×dup1" in error.message)
assertTrue("2×dup2" in error.message)
```

Reply:
![image](https://user-images.githubusercontent.com/101754/54641186-cb496480-4a4e-11e9-9566-05ae9b63f554.png)